internal/cssc/cssc.go

```diff
@@ -24,19 +27,42 @@ import (
 	"oras.land/oras-go/v2/registry/remote/retry"
 )
 
+type TagConvention string
+
+const (
+	Incremental TagConvention = "incremental"
+	Floating    TagConvention = "floating"
+)
+
+func (tc TagConvention) IsValid() error {
+	switch tc {
+	case Incremental, Floating:
+		return nil
+	}
+	return errors.New("tag-convention should be either incremental or floating")
+}
+
+var (
+	floatingTagRegex    = regexp.MustCompile(`^(.+)-patched`)             // Matches strings ending with -patched
+	incrementalTagRegex = regexp.MustCompile(`^(.+)-([1-9]\d{0,2})$`)     // Matches strings ending with -1 to -999
```

Comment:
This won't match things like 001. This may be intentional as well, but pointing it out just in case.

Reply:
yes, correct

Reply:
Isn't it more likely though for customers to use -1 accidentally than say -001? It sounds like -001 would reduce the likelihood of tag collisions

Reply:
The agreed upon convention right now with dependabot team is -1 to -999. There are more proposals in discussion like 
-patch-1 but the plan rn is to move forward with the current approach.

Reply:
Okay sure. I think we will run into issues, but if that's the agreed upon convention lets move forward